BREAKING CHANGE: Rename Scope to ExecutionPolicyScope

TraGicCode opened the pull request:
This Fixes #14 

Explanation is detailed within the linked issue above.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)
<!-- Reviewable:end -->

johlju commented:
@TraGicCode Merged PR #16. Please rebase this PR and resolve the merge conflicts. Please use `git rebase` (and not `git pull` or `git merge`).

TraGicCode commented:
Done.

---

Review status: 0 of 4 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-:-LAdSq0gE5FADBg29n-7:b-896fix)*
<!-- Sent from Reviewable.io -->

johlju commented:
I merged the other PR, so you need to rebase this one again, sorry 🙂

TraGicCode commented:
I'm on this merge conflict now

johlju commented:
Ah great! I will review tomorrow. Getting late here. 😄

TraGicCode commented:
Rebased and fixed conflicts.

johlju commented:
Review status: 0 of 4 files reviewed at latest revision, all discussions resolved.

---

*[a discussion](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAhB0jW6Y3BzFRpkWTp:-LAhB0jX2GZE4U8S2Atn:be2ptwc) (no related file):*
Even if this is a bug, we should make this a breaking change since we are renaming the parameter:
https://github.com/PowerShell/DscResources/blob/a923a5f1cda20bd69046095da22c3ba062644922/Maintainers.md#breaking-changes

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAhJMx10m7eSoeS_sVw:-LAhJMx10m7eSoeS_sVx:be6bz7a) (no related file):*
Would you mind adding another example of using ScopeLevel parameter?

---

*[README.md, line 47 at r1](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAhIkwS4EiuFQoj69ve:-LAhIkwT5nwEo3ftQlyz:b2rhlx3) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/9cf99f58b43c3ff0615a5a3a1368cbb96c88dfe3/README.md#L47)):*
> ```Markdown
>   * Fixed Markdown Linting issues as well.
> * [Michael Fyffe (@TraGicCode)](https://github.com/TraGicCode): Enabled Code Coverage Support ([issue #18](https://github.com/PowerShell/xPowerShellExecutionPolicy/issues/18)).
> * [Michael Fyffe (@TraGicCode)](https://github.com/TraGicCode): Renamed the 'Scope' Parameter to 'ScopeLevel' since errors are thrown when the MOF file is parsed because 'Scope' is a reserved keyword in the DMTF Specification. ([issue #14](https://github.com/PowerShell/xPowerShellExecutionPolicy/issues/14)).
> ```

Should we add 'BREAKING CHANGE:' label before the name, so it's clearly visible?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

TraGicCode commented:
*[README.md, line 47 at r1](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAhIkwS4EiuFQoj69ve:-LAmS_eQ5MCwO0cF-lmm:blu8fdt) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/9cf99f58b43c3ff0615a5a3a1368cbb96c88dfe3/README.md#L47)):*
<details><summary><i>Previously, TraGicCode (Michael Fyffe) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

FYI, Here is what my repos do which is auto-generated by the changelog generator ruby gem

https://github.com/TraGicCode/tragiccode-wsusserver/blob/master/CHANGELOG.md

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

kilasuit commented on DSCResources/MSFT_xPowerShellExecutionPolicy/MSFT_xPowerShellExecutionPolicy.psm1:
```diff
@@ -26,13 +26,13 @@ function Get-TargetResource
         [Parameter()]
```
I'd suggest making the parameter have an Alias to Scope here which I believe this should then mitigate it being a breaking change
You can do this with the Alias Parameter Attribute

kilasuit replied:
Of course you would need to add this to Set & Test functions as well.

TraGicCode replied:
@kilasuit ,

Very interesting idea.  I tested this and it seems to still work as intended and fix the bug that caused this PR to open.

@johlju What are your thoughts?  I'm okay with doing this.

TraGicCode replied:
Any ideas if there is something that needs to be added to the schema.mof if an alias is added?

johlju replied:
No idea honestly - I have never seen it being used. @kilasuit do you know?

johlju replied:
@TraGicCode I don't think it is possible to use alias. That would mean Get-TargetResource would be able to choose if it should return as Scope or ScopeLevel. I would be cool though if it worked.

TraGicCode commented:
*[a discussion](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAhB0jW6Y3BzFRpkWTp:-LAmUrWX5g75mC6rqCbG:brxp9sq) (no related file):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Even if this is a bug, we should make this a breaking change since we are renaming the parameter:
https://github.com/PowerShell/DscResources/blob/a923a5f1cda20bd69046095da22c3ba062644922/Maintainers.md#breaking-changes
</blockquote></details>

Sounds good.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

TraGicCode commented:
*[a discussion](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAhJMx10m7eSoeS_sVw:-LAmUuo73hBzTpopRgfg:b-pyeaxo) (no related file):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Would you mind adding another example of using ScopeLevel parameter?
</blockquote></details>

I just updated the existing example if this seems good for you.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 1 of 1 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[a discussion](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAhJMx10m7eSoeS_sVw:-LAmbUOO9kxNl8lRKo96:b-6yf3of) (no related file):*
<details><summary><i>Previously, TraGicCode (Michael Fyffe) wrote…</i></summary><blockquote>

I just updated the existing example if this seems good for you.
</blockquote></details>

Yes that is perfect, much better! But maybe we should have a second example without ScopeLevel too? My though is so it's clear what the difference is using ScopeLevel and not.

---

*[README.md, line 47 at r1](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAhIkwS4EiuFQoj69ve:-LAma9AaCr5O64W-U8FH:b9fappd) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/9cf99f58b43c3ff0615a5a3a1368cbb96c88dfe3/README.md#L47)):*
<details><summary><i>Previously, TraGicCode (Michael Fyffe) wrote…</i></summary><blockquote>

FYI, Here is what my repos do which is auto-generated by the changelog generator ruby gem

https://github.com/TraGicCode/tragiccode-wsusserver/blob/master/CHANGELOG.md
</blockquote></details>

I agree. SqlServerDsc was a first attempt doing this. I'm good moving it to the end, if it is better we should do that in SqlServerDsc as well. 

Maybe like this? Or is 'Resolved by' to much information?

BREAKING CHANGE: Renamed the 'Scope' Parameter to 'ScopeLevel' since errors are thrown when the MOF file is parsed because 'Scope' is a reserved keyword in the DMTF Specification ([issue #14](https://github.com/PowerShell/xPowerShellExecutionPolicy/issues/14)). Resolved by [Michael Fyffe (@TraGicCode)](https://github.com/TraGicCode).

```
BREAKING CHANGE: Renamed the 'Scope' Parameter to 'ScopeLevel' since errors are thrown when the MOF file is parsed because 'Scope' is a reserved keyword in the DMTF Specification ([issue #14](https://github.com/PowerShell/xPowerShellExecutionPolicy/issues/14)). Resolved by [Michael Fyffe (@TraGicCode)](https://github.com/TraGicCode).
```

---

*[DSCResources/MSFT_xPowerShellExecutionPolicy/MSFT_xPowerShellExecutionPolicy.psm1, line 26 at r2](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-L9uIVWqQ-gSshGqKXuC-r2-26:-LAm_X5W0qUZuuujkv14:b-w4i7o) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/1fb2dc66a7f4dc27b307375df0c126146f313472/DSCResources/MSFT_xPowerShellExecutionPolicy/MSFT_xPowerShellExecutionPolicy.psm1#L26)):*
<details><summary><i>Previously, TraGicCode (Michael Fyffe) wrote…</i></summary><blockquote>

@kilasuit ,

Very interesting idea.  I tested this and it seems to still work as intended and fix the bug that caused this PR to open.

@johlju What are your thoughts?  I'm okay with doing this.
</blockquote></details>

Sure I'm good with it. Just add tests that tests both alias an the real parameter name :smile:

---

*[Examples/1-SetPowerShellExecutionPolicy.ps1, line 3 at r3](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAmayEoB-pO0XQwmeDF:-LAmayEoB-pO0XQwmeDG:b8yeqy4) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/4b09a11ad31615f98807ca19bacea32932257992/Examples/1-SetPowerShellExecutionPolicy.ps1#L3)):*
> ```PowerShell
> <#
> .EXAMPLE
>     This example shows how to configure powershell's execution policy.
> ```

'...for the specified scope.' maybe?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

TraGicCode commented:
*[README.md, line 47 at r1](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAhIkwS4EiuFQoj69ve:-LAuDWd29jdWNx7cHrUc:b9natmm) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/9cf99f58b43c3ff0615a5a3a1368cbb96c88dfe3/README.md#L47)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

I agree. SqlServerDsc was a first attempt doing this. I'm good moving it to the end, if it is better we should do that in SqlServerDsc as well. 

Maybe like this? Or is 'Resolved by' to much information?

BREAKING CHANGE: Renamed the 'Scope' Parameter to 'ScopeLevel' since errors are thrown when the MOF file is parsed because 'Scope' is a reserved keyword in the DMTF Specification ([issue #14](https://github.com/PowerShell/xPowerShellExecutionPolicy/issues/14)). Resolved by [Michael Fyffe (@TraGicCode)](https://github.com/TraGicCode).

```
BREAKING CHANGE: Renamed the 'Scope' Parameter to 'ScopeLevel' since errors are thrown when the MOF file is parsed because 'Scope' is a reserved keyword in the DMTF Specification ([issue #14](https://github.com/PowerShell/xPowerShellExecutionPolicy/issues/14)). Resolved by [Michael Fyffe (@TraGicCode)](https://github.com/TraGicCode).
```

</blockquote></details>

Hmm, i feel like it's implied and the resolved by is redundant.  You okay with keeping it the way you have and without the Resolved by?

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

TraGicCode commented:
*[DSCResources/MSFT_xPowerShellExecutionPolicy/MSFT_xPowerShellExecutionPolicy.psm1, line 26 at r2](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-L9uIVWqQ-gSshGqKXuC-r2-26:-LAuDkR_AhduiWlDanUB:bujru7) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/1fb2dc66a7f4dc27b307375df0c126146f313472/DSCResources/MSFT_xPowerShellExecutionPolicy/MSFT_xPowerShellExecutionPolicy.psm1#L26)):*
<details><summary><i>Previously, TraGicCode (Michael Fyffe) wrote…</i></summary><blockquote>

I'm okay with leaving this as is unless @kilasuit has any objections.
</blockquote></details>

If he doesn't respond i'm okay with leaving as is.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

TraGicCode commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[Examples/1-SetPowerShellExecutionPolicy.ps1, line 3 at r3](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAmayEoB-pO0XQwmeDF:-LAuEl4c6hjmCPGnJSPj:b-896fix) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/4b09a11ad31615f98807ca19bacea32932257992/Examples/1-SetPowerShellExecutionPolicy.ps1#L3)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

'...for the specified scope.' maybe?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[README.md, line 47 at r1](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LAhIkwS4EiuFQoj69ve:-LAvwIkh1rvXBqtOqvtm:b6esetf) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/9cf99f58b43c3ff0615a5a3a1368cbb96c88dfe3/README.md#L47)):*
<details><summary><i>Previously, TraGicCode (Michael Fyffe) wrote…</i></summary><blockquote>

Hmm, i feel like it's implied and the resolved by is redundant.  You okay with keeping it the way you have and without the Resolved by?
</blockquote></details>

I'm okay with that.

---

*[DSCResources/MSFT_xPowerShellExecutionPolicy/MSFT_xPowerShellExecutionPolicy.psm1, line 26 at r2](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-L9uIVWqQ-gSshGqKXuC-r2-26:-LAvw39C217MyCDaaXKN:b6bhitp) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/1fb2dc66a7f4dc27b307375df0c126146f313472/DSCResources/MSFT_xPowerShellExecutionPolicy/MSFT_xPowerShellExecutionPolicy.psm1#L26)):*
<details><summary><i>Previously, TraGicCode (Michael Fyffe) wrote…</i></summary><blockquote>

If he doesn't respond i'm okay with leaving as is.
</blockquote></details>

Leave as-is if you doesn't here from @kilasuit.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

johlju commented:
@TraGicCode Since we don't have an integration tests for this resource, have you run this in a lab environment so you have seen this works?

johlju commented:
@TraGicCode let me know the result as soon as you have had a chance to test, no hurry from my end.

johlju commented:
It works for me. But zipped your code into a module and put it onto the "share" so the server downloaded it.

@TraGicCode Would you be up to writing an integration test? if so I think copying a test from SqlServerDsc would be easiest. 

```
VERBOSE: [SQLTEST]: LCM:  [ Start  Set      ]
VERBOSE: [SQLTEST]:                            [DSCEngine] Module xPowerShellExecutionPolicy_2.0.0.0.zip downloaded.
VERBOSE: [SQLTEST]:                            [DSCEngine] Resource xPowerShellExecutionPolicy version 2.0.0.0 installed.
VERBOSE: [SQLTEST]: LCM:  [ Start  Resource ]  [[xPowerShellExecutionPolicy]ExecutionPolicy]
VERBOSE: [SQLTEST]: LCM:  [ Start  Test     ]  [[xPowerShellExecutionPolicy]ExecutionPolicy]
VERBOSE: [SQLTEST]: LCM:  [ End    Test     ]  [[xPowerShellExecutionPolicy]ExecutionPolicy]  in 0.0470 seconds.
VERBOSE: [SQLTEST]: LCM:  [ Skip   Set      ]  [[xPowerShellExecutionPolicy]ExecutionPolicy]
VERBOSE: [SQLTEST]: LCM:  [ End    Resource ]  [[xPowerShellExecutionPolicy]ExecutionPolicy]
```

TraGicCode commented:
Hey @johlju ,

What is wrong with the approach i was trying?

Also i don't mind trying to add an integration test.

johlju commented:
@TraGicCode maybe it needs be the correct folder layout?

```
C:\PROGRAM FILES\WINDOWSPOWERSHELL\MODULES\XPOWERSHELLEXECUTIONPOLICY
└───2.0.0.0
    ├───DSCResources
    │   └───MSFT_xPowerShellExecutionPolicy
    ├───Examples
    └───Tests
        └───Unit
```

johlju commented:
@TraGicCode I will wait for an integration test then. It would be awesome to be able to close that issue as well 😄

TraGicCode commented:
Hey @johlju ,

I added a basic integration test from the template the community provides.  Is there anything you would change about the integration test that i put in place?

johlju commented:
Reviewed 5 of 5 files at r5, 2 of 2 files at r6, 2 of 2 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Tests/Integration/MSFT_xPowerShellExecutionPolicy.config.ps1, line 4 at r7](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LBWEQK_B0PL4ybwXwzz:-LBWEQK_B0PL4ybwXx--:biiz7u6) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/90be2665a51adaa8e70746ba6cfd9af0d17d3e01/Tests/Integration/MSFT_xPowerShellExecutionPolicy.config.ps1#L4)):*
> ```PowerShell
> # Integration Test Config Template Version: 1.0.0
> configuration MSFT_xPowerShellExecutionPolicy_config {
>     # TODO: Modify ModuleName (e.g. xNetworking)
> ```

Please remove todo comment here.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

TraGicCode commented:
Done.  Thanks for the help on this 🎉

TraGicCode commented:
Review status: 8 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[Tests/Integration/MSFT_xPowerShellExecutionPolicy.config.ps1, line 4 at r7](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-LBWEQK_B0PL4ybwXwzz:-LBX0_QOEh6i41l1001g:b-896fix) ([raw file](https://github.com/powershell/xpowershellexecutionpolicy/blob/90be2665a51adaa8e70746ba6cfd9af0d17d3e01/Tests/Integration/MSFT_xPowerShellExecutionPolicy.config.ps1#L4)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please remove todo comment here.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15)*
<!-- Sent from Reviewable.io -->

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/15#-:-LBfADiA7UFabYavgoaB:bnfp4nl)*
<!-- Sent from Reviewable.io -->

johlju commented:
@TraGicCode Awesome work on this! Thanks for closing so many issue!!
